app/assets/javascripts/report-a-problem-form.js

```diff
@@ -0,0 +1,83 @@
+(function() {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  var ReportAProblemForm = function($form) {
+    this.$form = $form;
+
+    $form.on('submit', $.proxy(this.submit, this));
+
+    this.appendHiddenContextInformation();
+  }
+
+  ReportAProblemForm.prototype.appendHiddenContextInformation = function() {
+    // form submission for reporting a problem
+    this.$form.append('<input type="hidden" name="javascript_enabled" value="true"/>');
+    this.$form.append($('<input type="hidden" name="referrer">').val(document.referrer || "unknown"));
+  };
+
+  ReportAProblemForm.prototype.hidePrompt = function() {
+    this.$form.find('.error-notification').remove();
+  };
+
+  ReportAProblemForm.prototype.disableSubmitButton = function() {
+    this.$form.find('.button').attr("disabled", true);
+  };
+
+  ReportAProblemForm.prototype.enableSubmitButton = function() {
+    this.$form.find('.button').attr("disabled", false);
+  };
+
+  ReportAProblemForm.prototype.promptUserToEnterValidData = function() {
+    this.enableSubmitButton();
+    var $prompt = $('<p class="error-notification">Please enter details of what you were doing.</p>');
+    this.$form.prepend($prompt);
+  };
+
+  ReportAProblemForm.prototype.handleError = function(jqXHR, status) {
+    if (status === 'error' || !jqXHR.responseText) {
+      if (jqXHR.status == 422) {
+        this.promptUserToEnterValidData();
+      } else {
+        this.triggerError();
+      }
+    }
+  };
+
+  ReportAProblemForm.prototype.setUrl = function() {
+    this.$form.find('input#url').val(window.location);
+  }
+
+  ReportAProblemForm.prototype.postFormViaAjax = function() {
+    $.ajax({
+      type: "POST",
+      url: "/contact/govuk/problem_reports",
+      dataType: "json",
+      data: this.$form.serialize(),
+      success: $.proxy(this.triggerSuccess, this),
```

Comment:
Any reason to use `proxy` over `Function.prototype.bind`? It would be good to move everything to `bind` where possible for consistency. In this case the `bind` polyfill is already loaded by the time this code will be run.